### PR TITLE
Remove legacy fee

### PR DESCRIPTION
**Pull request by vkgnosis**
https://github.com/cowprotocol/services/issues/119

### CODE REVIEW

**Comment by nlordell** [comment]
recheck

File: src/gas_price.rs [hunk 1]
```diff
@@ -1,122 +1,44 @@
 use anyhow::{anyhow, Result};
-/// Gas price received from the gas price estimators.
 use serde::Serialize;
 
+/// EIP1559 gas price
```
Comment (vkgnosis) [inline src/gas_price.rs:4]: This is the main change

File: src/nativegasestimator.rs [hunk 2]
```diff
@@ -113,14 +113,13 @@ impl NativeGasEstimator {
...
-                                gas_price.cap().max(gas_price.base_fee())
-                                    * params.bump_cap_coefficient,
-                            ),
-                        )
+                    .map(|(time_limit, mut gas_price)| {
+                        // TODO: Double check with Dusan whether this makes sense. I haven't changed
+                        // the logic but this feels weird. (same in `handle` below)
+                        gas_price.max_fee_per_gas =
+                            gas_price.max_fee_per_gas.max(gas_price.base_fee_per_gas)
+                                * params.bump_cap_coefficient;
```
[the remark this answers is not in the document]
Comment (sunce86) [reply src/nativegasestimator.rs:121]: Yup, that's it.

**Review by sunce86 — APPROVED** [review]
lgtm

**Pull request merged**